scriptmodules/supplementary/skyscraper.sh

```diff
@@ -0,0 +1,696 @@
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="skyscraper"
+rp_module_desc="Scraper for EmulationStation by Lars Muldjord"
+rp_module_licence="GPLv3.0 https://raw.githubusercontent.com/muldjord/skyscraper/master/LICENSE"
+rp_module_section="opt"
+
+function depends_skyscraper() {
+    getDepends qt5-default p7zip-full
+}
+
+function sources_skyscraper() {
+    local latest_ver
+    latest_ver=$(_latest_ver_skyscraper)
+
+    gitPullOrClone "$md_build" "https://github.com/muldjord/skyscraper" "$latest_ver"
+}
+
+function build_skyscraper() {
+    qmake
+    make
+    md_ret_require="$md_build/Skyscraper"
+}
+
+function install_skyscraper() {
+    md_ret_files=(
+    'Skyscraper'
+    'LICENSE'
+    'README.md'
+    )
+}
+
+# Purge all Skyscraper caches
+function _purge_skyscraper() {
+    local platform
+
+    while read platform; do
+        # Find any folders underneath the Local DB cache and clear them
+        _clear_platform_skyscraper "$platform"
+    done < <(find "$configdir/all/skyscraper/dbs" -maxdepth 1 -mindepth 1 -type d -exec basename {} \;)
+}
+
+function _clear_platform_skyscraper() {
+    local platform="$1"
+
+    [[ ! -d "$configdir/all/skyscraper/dbs/$platform" ]] && return
+
+    # Remove any folder underneath the platform, try to keep any customized files
+    pushd "$configdir/all/skyscraper/dbs/$platform"
+    find . -maxdepth 1 -mindepth 1 -type d -print0 -exec rm -fr {} \;
+    rm -f db.xml
+    popd
+
+}
+# Purge by cached platform
+function _purge_platform_skyscraper() {
+    local options=()
+    local system
+
+    while read system; do
+        # If there is no 'db.xml' file underneath the folder, skip it, it means folder is empty
+        [[ ! -f "$configdir/all/skyscraper/dbs/$system/db.xml" ]] && continue
+
+        # Get the size on disk of the system and show it in the select list
+        local size=$(du -sh  "$configdir/all/skyscraper/dbs/$system" | cut -f1)
+        options+=("$system" "$size" OFF)
+    done < <(find "$configdir/all/skyscraper/dbs" -maxdepth 1 -mindepth 1 -type d -exec basename {} \;)
+
+    # If not folders are found, show an info message instead of the selection list
+    if [[ ${#options[@]} -eq 0 ]] ; then
+        printMsgs "dialog" "Nothing to delete ! No cached systems found in \n$configdir/all/skyscraper/dbs "
+        return
+    fi
+
+    local cmd=(dialog --backtitle "$__backtitle" --radiolist "Select platform to purge" 20 60 12)
+    local platform=$("${cmd[@]}" "${options[@]}" 2>&1 >/dev/tty)
+
+    # Exit if no platform chosen
+    [[ -z "$platform" ]] && return
+
+    _clear_platform_skyscraper "$platform"
+}
+
+
+function _get_ver_skyscraper() {
+    [[ -f "$md_inst/Skyscraper" ]] && "$md_inst/Skyscraper" -h | grep 'Running Skyscraper'  | cut -d' '  -f 3 2>/dev/null
```

Comment:
maybe use `if` here so it's not all on one line as it's rather long.

Reply:
It was supposed to be a one liner to get the version, but I think it's clearer to use an `if`. Modified accordingly.